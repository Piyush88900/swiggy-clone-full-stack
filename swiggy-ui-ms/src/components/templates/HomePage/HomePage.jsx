/**
 *
 * HomePage
 *
 */
import React from 'react';
import { Grid, Box, Container } from '@material-ui/core';
import Layout from '../Layout';
import Home from '../../organisms/Home';
import Link from '../../atoms/Link';

const HomePage = () => {
  return (
    <Layout>
      <Container>
        <Grid item xs={12}>
          <Box my={2}>
            <Link href="/dashboard">Dashboard</Link>
          </Box>
        </Grid>
        <Grid item xs={12}>
          <Home />
        </Grid>
      </Container>
    </Layout>
  );
};

export default HomePage;
