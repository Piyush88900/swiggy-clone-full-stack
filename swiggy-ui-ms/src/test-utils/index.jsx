import React from 'react';
import PropTypes from 'prop-types';
import { render } from '@testing-library/react';
import { MockedProvider } from '@apollo/client/testing';
import { ThemeProvider } from '@material-ui/core/styles';
import { LabelProvider } from '../stores/labels';
import { theme } from '../theme';

const AllTheProviders = ({ children, mocks = [] }) => {
  return (
    <MockedProvider mocks={mocks} addTypename={false}>
      <ThemeProvider theme={theme}>
        <LabelProvider>{children}</LabelProvider>
      </ThemeProvider>
    </MockedProvider>
  );
};

AllTheProviders.propTypes = {
  children: PropTypes.node.isRequired,
  mocks: PropTypes.arrayOf(PropTypes.shape({})),
};

AllTheProviders.defaultProps = {
  mocks: [],
};

const customRender = (ui, options) => {
  const { wrapperProps = {}, ...customOptions } = options || {};
  return render(ui, {
    wrapper: (props) => {
      return <AllTheProviders {...props} {...wrapperProps} />;
    },
    ...customOptions,
  });
};

// re-export everything
export * from '@testing-library/react';
export * from './withTestRouter';

// override render method
export { customRender as render };
