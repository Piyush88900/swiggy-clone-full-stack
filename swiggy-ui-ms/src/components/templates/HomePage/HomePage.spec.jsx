import React from 'react';
import { render, withTestRouter } from '../../../test-utils';

import HomePage from '.';

describe('HomePage', () => {
  it('should render correctly', () => {
    const { asFragment } = render(withTestRouter(<HomePage />));
    expect(asFragment()).toMatchSnapshot();
  });
});
