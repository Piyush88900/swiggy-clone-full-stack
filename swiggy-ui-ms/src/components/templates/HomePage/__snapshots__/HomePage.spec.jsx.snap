// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage should render correctly 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-root-1"
  >
    <header
      class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
    >
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg"
      >
        <div
          class="MuiToolbar-root MuiToolbar-regular"
        >
          <h6
            class="MuiTypography-root makeStyles-title-3 MuiTypography-h6"
          >
            Hiring & Staffing
          </h6>
        </div>
      </div>
    </header>
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <div
          class="MuiBox-root MuiBox-root-4"
        >
          <a
            class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
            href="/dashboard"
          >
            Dashboard
          </a>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        ...loading content
      </div>
    </div>
  </div>
</DocumentFragment>
`;
